### ISSUE TITLE
500/502/504 Server Error while using BertTokenizer

### ISSUE THREAD

**Issue by citrovin** [huggingface/tokenizers#865]
Hi,

I am using the class BertTokenizer (with the scivocab file: 'allenai/scibert_scivocab_uncased') to tokenize my text for the pre-training of my language model. I have tried starting the pre-training a few times now. However, I always get a Server Error at different stages of my training (my dataset has about 8000 samples and I do not shuffle my dataset and therefore the pre-training should take about 9 hours). For example, during the first training an error occured after about 1227 samples and during the second training it occured after 4022 samples. Since the error is not consistent at a specific sample, I therefore I do not assume that there is a problem with my code. However, my code looks as follows, where the variable `text` represents a sample of my dataset and I call the tokenizer in a loop:
```
    # Load pre-trained model tokenizer (vocabulary)
    tokenizer = BertTokenizer.from_pretrained('allenai/scibert_scivocab_uncased')

    # create the input_ids, token_type_ids, attention_mask
    batch = tokenizer(text, max_length=512, padding='max_length', truncation=True)
```

Here are the actual errors I get during training:
1. 504 Server Error: Gateway Time-out for url: https://huggingface.co/api/models/allenai/scibert_scivocab_uncased
2. 502 Server Error: Bad Gateway for url: https://huggingface.co/allenai/scibert_scivocab_uncased/resolve/main/special_tokens_map.json
3. 500 Server Error: Internal Server Error for url: https://huggingface.co/allenai/scibert_scivocab_uncased/resolve/main/vocab.txt
4. 502 Server Error: Bad Gateway for url: https://huggingface.co/allenai/scibert_scivocab_uncased/resolve/main/tokenizer.json

I was wondering if you have had this problem before and if there is a work around?

If you need more information regarding the problem, please do not hesitate to contact me. I would really appreciate your help! 

Kind regards

citrovin

**Comment by Narsil** [comment]
Hi @citrovin ,

I think you should load the tokenizer only once if you want to avoid issues. It's also going to be much faster.
If you keep loading in the loop, then many errors might start popping as you might be considered starting DOS attempts by automated checks.

```python
tokenizer = AutoTokenizer.from_pretrained(....)

for example in examples:
    batch = tokenizer(text, ...)
```
should be much better. And you shouldn't see anymore errors. Can you confirm?

**Comment by citrovin** [comment]
Thank you very much! I tested it twice and it definitely works better now.
